### DISCUSSION TITLE
How to expand all macros of a script?

### DISCUSSION THREAD

**Discussion by MaxGyver83** [janet-lang/janet#1423] [Q&A] [answered]
Is there a simple way to expand all macros of a script? Like `gcc -E`?

**Comment by CosmicToast** [comment, marked as answer]
You can just parse the source script and macex on it.
Though note that it won't be super readable, and won't apply to modules (I'm pretty sure anyway).
Here's an example:
```janet
# test.janet
(do
  (print "hi")
  (%= 1 2))
```

```janet
# test2.janet
(def src (parse-all (slurp "test.janet")))
(pp src)
(pp (macex src))
```

```sh
janet test2.janet
# => @[(do (print "hi") (%= 1 2))]
# => @[(do (print "hi") (set 1 (<function %> 1 2)))]
```

**Reply by MaxGyver83** [reply]
Cool, thank you! Easier than I thought.
